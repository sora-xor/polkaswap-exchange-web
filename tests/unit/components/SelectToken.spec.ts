// import { shallowMount, createLocalVue } from '@vue/test-utils'
// import Vuex from 'vuex'

// import SelectToken from '@/components/SelectToken.vue'
// import { tokens } from '../../mocks/tokens'
// import { SoramitsuElementsImport, TranslationMock } from '../../utils'

// const localVue = createLocalVue()
// localVue.use(Vuex)
// SoramitsuElementsImport(localVue)

// describe('SelectToken.vue', () => {
//   let actions
//   let getters
//   let store

//   beforeEach(() => {
//     TranslationMock(SelectToken)

//     actions = {
//       getTokens: jest.fn()
//     }

//     getters = {
//       tokens: () => tokens
//     }

//     store = new Vuex.Store({
//       actions,
//       getters
//     })
//   })

//   it('should renders correctly', () => {
//     const wrapper = shallowMount(SelectToken, {
//       localVue,
//       store,
//       propsData: {
//         visible: true,
//         title: 'Select a token'
//       }
//     })
//     expect(wrapper.element).toMatchSnapshot()
//   })

//   // TODO: fix the test
//   it('should close correctly', async () => {
//     const wrapper = shallowMount(SelectToken, {
//       localVue,
//       store,
//       propsData: {
//         visible: true,
//         title: 'Select a token'
//       }
//     })
//     const token = wrapper.find('.token-item')
//     token.trigger('click')
//     expect(wrapper.emitted('select')).toBeTruthy()
//     expect(wrapper.emitted('close')).toBeTruthy()
//   })
// })

describe('SelectToken test', () => {
  test('Temporary test', () => {
    expect(true).toEqual(true);
  });
});
