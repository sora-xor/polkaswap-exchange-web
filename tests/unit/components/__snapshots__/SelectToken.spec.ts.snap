// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectToken.vue should renders correctly 1`] = `
<dialog-base-stub
  customclass="token-select"
  title="Select a token"
>
  <s-input-stub
    accept=""
    autocomplete="off"
    borderradius="mini"
    class="token-search"
    form=""
    label=""
    name=""
    placeholder=""
    prefix="el-icon-search"
    size="medium"
    suffix=""
    tabindex=""
    type="text"
    value=""
  />
   
  <div
    class="token-list"
  >
    <div
      class="token-item"
    >
      <s-col-stub
        offset="0"
        pull="0"
        push="0"
        span="12"
      >
        <s-row-stub
          align="middle"
          flex="true"
          gutter="0"
          justify="start"
        >
          <!---->
           
          <div>
            <div
              class="token-item__name"
            >
              Sora (XOR)
            </div>
          </div>
        </s-row-stub>
      </s-col-stub>
       
      <div>
        <span
          class="token-item__amount"
        >
          -
        </span>
      </div>
    </div>
    <div
      class="token-item"
    >
      <s-col-stub
        offset="0"
        pull="0"
        push="0"
        span="12"
      >
        <s-row-stub
          align="middle"
          flex="true"
          gutter="0"
          justify="start"
        >
          <!---->
           
          <div>
            <div
              class="token-item__name"
            >
              Kusama (KSM)
            </div>
          </div>
        </s-row-stub>
      </s-col-stub>
       
      <div>
        <span
          class="token-item__amount"
        >
          -
        </span>
      </div>
    </div>
    <div
      class="token-item"
    >
      <s-col-stub
        offset="0"
        pull="0"
        push="0"
        span="12"
      >
        <s-row-stub
          align="middle"
          flex="true"
          gutter="0"
          justify="start"
        >
          <!---->
           
          <div>
            <div
              class="token-item__name"
            >
              Kusama (KSM1)
            </div>
          </div>
        </s-row-stub>
      </s-col-stub>
       
      <div>
        <span
          class="token-item__amount"
        >
          -
        </span>
      </div>
    </div>
    <div
      class="token-item"
    >
      <s-col-stub
        offset="0"
        pull="0"
        push="0"
        span="12"
      >
        <s-row-stub
          align="middle"
          flex="true"
          gutter="0"
          justify="start"
        >
          <!---->
           
          <div>
            <div
              class="token-item__name"
            >
              Kusama (KSM2)
            </div>
          </div>
        </s-row-stub>
      </s-col-stub>
       
      <div>
        <span
          class="token-item__amount"
        >
          -
        </span>
      </div>
    </div>
    <div
      class="token-item"
    >
      <s-col-stub
        offset="0"
        pull="0"
        push="0"
        span="12"
      >
        <s-row-stub
          align="middle"
          flex="true"
          gutter="0"
          justify="start"
        >
          <!---->
           
          <div>
            <div
              class="token-item__name"
            >
              Kusama (KSM3)
            </div>
          </div>
        </s-row-stub>
      </s-col-stub>
       
      <div>
        <span
          class="token-item__amount"
        >
          -
        </span>
      </div>
    </div>
    <div
      class="token-item"
    >
      <s-col-stub
        offset="0"
        pull="0"
        push="0"
        span="12"
      >
        <s-row-stub
          align="middle"
          flex="true"
          gutter="0"
          justify="start"
        >
          <!---->
           
          <div>
            <div
              class="token-item__name"
            >
              Kusama (KSM4)
            </div>
          </div>
        </s-row-stub>
      </s-col-stub>
       
      <div>
        <span
          class="token-item__amount"
        >
          -
        </span>
      </div>
    </div>
    <div
      class="token-item"
    >
      <s-col-stub
        offset="0"
        pull="0"
        push="0"
        span="12"
      >
        <s-row-stub
          align="middle"
          flex="true"
          gutter="0"
          justify="start"
        >
          <!---->
           
          <div>
            <div
              class="token-item__name"
            >
              Kusama (KSM5)
            </div>
          </div>
        </s-row-stub>
      </s-col-stub>
       
      <div>
        <span
          class="token-item__amount"
        >
          -
        </span>
      </div>
    </div>
    <div
      class="token-item"
    >
      <s-col-stub
        offset="0"
        pull="0"
        push="0"
        span="12"
      >
        <s-row-stub
          align="middle"
          flex="true"
          gutter="0"
          justify="start"
        >
          <!---->
           
          <div>
            <div
              class="token-item__name"
            >
              Kusama (KSM6)
            </div>
          </div>
        </s-row-stub>
      </s-col-stub>
       
      <div>
        <span
          class="token-item__amount"
        >
          -
        </span>
      </div>
    </div>
    <div
      class="token-item"
    >
      <s-col-stub
        offset="0"
        pull="0"
        push="0"
        span="12"
      >
        <s-row-stub
          align="middle"
          flex="true"
          gutter="0"
          justify="start"
        >
          <!---->
           
          <div>
            <div
              class="token-item__name"
            >
              Kusama (KSM7)
            </div>
          </div>
        </s-row-stub>
      </s-col-stub>
       
      <div>
        <span
          class="token-item__amount"
        >
          -
        </span>
      </div>
    </div>
    <div
      class="token-item"
    >
      <s-col-stub
        offset="0"
        pull="0"
        push="0"
        span="12"
      >
        <s-row-stub
          align="middle"
          flex="true"
          gutter="0"
          justify="start"
        >
          <!---->
           
          <div>
            <div
              class="token-item__name"
            >
              Ether (ETH)
            </div>
          </div>
        </s-row-stub>
      </s-col-stub>
       
      <div>
        <span
          class="token-item__amount"
        >
          -
        </span>
      </div>
    </div>
  </div>
</dialog-base-stub>
`;
